src/watertap_contrib/reflo/unit_models/zero_order/forward_osmosis_zo.py

```diff
@@ -0,0 +1,886 @@
+            )
+
+        @self.Constraint(doc="Product water separated at designed temperature")
+        def eq_product_temp(b):
+            return (
+                b.product_props[0].temperature
+                == b.regeneration_temp - b.separator_temp_loss
+            )
+
+    def initialize_build(
+        blk,
+        state_args=None,
+        outlvl=idaeslog.NOTSET,
+        solver=None,
+        optarg=None,
+    ):
+        """
+        General wrapper for initialization routines
+
+        Keyword Arguments:
+            state_args : a dict of arguments to be passed to the property
+                         package(s) to provide an initial state for
+                         initialization (see documentation of the specific
+                         property package) (default = {}).
+            outlvl : sets output level of initialization routine
+            optarg : solver options dictionary object (default=None)
+            solver : str indicating which solver to use during
+                     initialization (default = None)
+
+        Returns: None
+        """
+        init_log = idaeslog.getInitLogger(blk.name, outlvl, tag="unit")
+        solve_log = idaeslog.getSolveLogger(blk.name, outlvl, tag="unit")
+
+        opt = get_solver(solver, optarg)
+        # ---------------------------------------------------------------------
+        flags = blk.feed_props.initialize(
+            outlvl=outlvl,
+            optarg=optarg,
+            solver=solver,
+            state_args=state_args,
+            hold_state=True,
+        )
+        init_log.info("Initialization Step 1 Complete.")
+        # ---------------------------------------------------------------------
+        # Initialize other state blocks
+        if state_args is None:
+            blk.state_args = state_args = {}
+            state_dict = blk.feed_props[
+                blk.flowsheet().config.time.first()
+            ].define_port_members()
+
+            for k in state_dict.keys():
+                if state_dict[k].is_indexed():
+                    state_args[k] = {}
+                    for m in state_dict[k].keys():
+                        state_args[k][m] = state_dict[k][m].value
+                else:
+                    state_args[k] = state_dict[k].value
+
+        # Initial guess of brine properties
+        state_args_brine = deepcopy(state_args)
+        for p, j in blk.brine_props.phase_component_set:
+            if p == "Liq" and j == "H2O":
+                state_args_brine["flow_mass_phase_comp"][(p, j)] = (
+                    state_args["flow_mass_phase_comp"][(p, j)]
+                    * (1 - blk.recovery_ratio / blk.nanofiltration_recovery_ratio)
+                    * pyunits.kg
+                    / pyunits.s
+                )
+
+        blk.brine_props.initialize(
+            outlvl=outlvl,
+            optarg=optarg,
+            solver=solver,
+            state_args=state_args_brine,
+        )
+
```

Comment:
```suggestion
init_log.info("Initialization of briner properties {}.".format(idaeslog.condition(res)))
```

Since there are so many state blocks in this model, it would be useful to add log messages for each of them imo.

Reply:
The messages are already scripted in the property package and will display if we set `outlvl = idaeslog.INFO_HIGH` (currently is `idaeslog.NOTSET`). I'll leave it as is unless you think we should change the significance of the log messages.